docs/sources/operations/storage/retention.md

```diff
@@ -3,7 +3,157 @@ title: Retention
 ---
 # Loki Storage Retention
 
-Retention in Loki is achieved through the [Table Manager](../table-manager/).
+Retention in Loki is achieved either through the [Table Manager](#table-manager) or the [Compactor](#Compactor).
+
+## Which one should you use ?
+
+Retention through the [Table Manager](../table-manager/) is achieved by relying on the object store TTL feature and will work for both [boltdb-shipper](../boltdb-shipper) store and chunk/index store. However retention through the [Compactor](../boltdb-shipper#compactor) is only supported when using [boltdb-shipper](../boltdb-shipper) store.
+
+Going forward the [Compactor](#Compactor) retention will become the default and long term supported one. While this retention is still **experimental** it supports more granular retention (per tenant and per stream) use cases.
+
+## Compactor
+
+As explained in the [Compactor](../boltdb-shipper#compactor) section, the compactor can deduplicate index entries but can also apply granular retention when activated.
+When applying retention with the Compactor, you can totally remove the [Table Manager](../table-manager/) from your Loki workload as it become totally unnecessary.
+
+> The compactor should always be run as a singleton. (single instance)
+
+### How it works ?
+
+Once activated, the Compactor will run a in loop that will apply compaction and retention at every `compaction_interval`. If it runs behind, the will executed as soon as possible.
+
+The execution is as follow:
+
+- First the compactor will download all available tables (previously uploaded by ingesters) for each day.
+- For each day and table it will:
+  - First compact the table into a single index file.
+  - Then traverse the whole index to mark chunks that need to be removed based on the tenant configuration.
+  - Marked chunks are directly removed from the index and their reference is saved into a file on disk.
+  - Finally re-upload the new modified index files.
+
+Retention is applying directly on the index, however chunks are not deleted directly. They actually will be deleted by the compactor asynchronously in another loop (sweeper).
+
+Marked chunks will only  be deleted after `retention_delete_delay` configured is expired because:
+
+1. boltdb-shipper indexes are refreshed from the shared store on components using it (querier and ruler) at a specific interval. This means deleting chunks instantly could lead to components still having reference to old chunks and so they could fails to execute queries. Having a delay allows for components to refresh their store and so remove gracefully their reference of those chunks.
+
+2. It gives you a short period to cancel chunks deletion in case of mistakes.
+
+Marker files (containing chunks to deletes), should be store on a persistent disk, since this is the sole reference to them anymore.
```

Comment:
I'm trying to explain why `retention_delete_delay` exist and how it should be configured.

Setting a too low delay could lead to unexpected behaviour. The default is 2h and I don't think it should ever be lower. Maybe we should leave it to that, since we don't yet have a way to rebuild back the index.

As for 2, someone could have made a wrong configuration, and the compactor could have ran, it's not easy to rebuild the index (we don't have out of the box solution) but if they ever had any super important data, it's not deleted. To cancel the deletion you just need to delete the marker files generated, they are store on the disk of their choice using `working_directory`. I think we should skip this part.

WDYT ? I still want to express why there's a `retention_delete_delay` and what it does.

Reply:
For now, let's fix some grammar, and consider in the future how to make this better. In the future, we should either explain how to cancel the deletion or provide a link within the docs to the place where it is explained.